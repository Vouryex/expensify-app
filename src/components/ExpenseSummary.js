import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
    </div>
  );
};

const mapStateToProps = state => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: expenses.length,
    expensesTotal: selectExpensesTotal(expenses),
  };
};

export default connect(mapStateToProps)(ExpenseSummary);
