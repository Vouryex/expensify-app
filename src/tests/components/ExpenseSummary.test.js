import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(
    <ExpenseSummary expensesCount={1} expensesTotal={30200} />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
  const wrapper = shallow(
    <ExpenseSummary expensesCount={3} expensesTotal={90500} />,
  );
  expect(wrapper).toMatchSnapshot();
});
