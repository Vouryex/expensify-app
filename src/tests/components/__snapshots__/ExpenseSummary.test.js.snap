// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling 
  $302.00
</div>
`;

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div>
  Viewing 
  3
   
  expenses
   totalling 
  $905.00
</div>
`;
